### ISSUE TITLE
Allow Promise.all for multiple fireEvent.update

### ISSUE THREAD

**Issue by spaceninja** [testing-library/eslint-plugin-testing-library#267]
`await-fire-event` says the following code is invalid because `fireEvent.update` must be awaited

```js
await Promise.all([
  fireEvent.update(isbn, book.isbn),
  fireEvent.update(title, book.title),
  fireEvent.update(authorFirst, book.authorFirst),
  fireEvent.update(authorLast, book.authorLast),
  fireEvent.update(rating, book.rating),
  fireEvent.update(length, book.length),
]);
```

Is there a reason using `Promise.all` shouldn't be allowed?

**Comment by Belco90** [comment]
Hi @spaceninja. Thanks for notifying this. Indeed, Promise.all should be a valid way of handling promises returned by async fire event methods. This was fixes couple of days ago when refactoring the rule for v4, you can see in [this PR](https://github.com/testing-library/eslint-plugin-testing-library/pull/265/files) it covers that case now. On v4 e are making sure all async methods are reported , so `await-fire-event` will consider Promise.all as valid, and will report variable references too.

However, v4 of the plugin is not finished yet. It's really close but still refactoring last bunch of rules. So I'm afraid you'll have to disable the rule for that line at the moment while v4 gets released.

**Comment by spaceninja** [comment]
No problem, thank you for the update! I'll disable the rule for now and I'm looking forward to v4. :)

**Comment by Belco90** [comment]
@spaceninja looking forward to it too! By the way, since my main FE framework/UI library is React (where fire event is sync): how often do you use `Promise.all` + fire event methods? Are you using `Promise.all` for wrapping other Vue Testing Library promises? What about `Promise.allSettled`?

**Comment by spaceninja** [comment]
I'm actually pretty new to testing in general. I'm using a side project to try to improve my skills. This was driven by me having those six `await` statements in a row and thinking "this is dumb, I don't need these to fire one at a time, I just need to know when they're all done." I looked at `Promise.allSettled` but I actually like the fact that `Promise.all` will break early if any of the promises fail, so `allSettled` didn't seem to be an improvement, unless I'm missing something.

**Comment by Belco90** [comment]
I see. I'm sorry this plugin wasn't helpful in that particular case then.

Nothing really interesting from `Promise.allSettled` but it technically is another valid way of handling promises, so I included it on v4 too. Not really sure anyone will use it, but I think the plugin must consider it anyway. Thanks for your feedback!

**Comment by Belco90** [comment]
This should be fixed on v4.0.0

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/eslint-plugin-testing-library/v/4.0.0)
- [GitHub release](https://github.com/testing-library/eslint-plugin-testing-library/releases/tag/v4.0.0)
